django/db/models/sql/subqueries.py

```diff
@@ -142,7 +142,13 @@ def add_update_fields(self, values_seq):
         that will be used to generate the UPDATE query. Might be more usefully
         called add_update_targets() to hint at the extra information here.
         """
-        self.values.extend(values_seq)
+        for field, model, val in values_seq:
+            if hasattr(val, 'resolve_expression'):
+                # Resolve expressions here so that annotations are no longer needed
+                val = val.resolve_expression(self, allow_joins=False, for_save=True)
+                if val.contains_aggregate:
+                    raise FieldError("Aggregate functions are not allowed in this query")
```

Comment:
I assumed that field error was covered by another test in the update code but it does not appear to be. It's not a new check, it's copied from `SQLUpdateCompiler` where I grabbed the resolve code from.

The error will be caught in it's original place in `SQLUpdateCompiler`, adding it here just caught it earlier in the process. I've added an extra test to cover it and dropped the `contains_aggregate` check here so that there's less duplicated code.